v03_pipeline/lib/misc/sample_ids.py

```diff
@@ -20,18 +18,23 @@ def remap_sample_ids(
     ignore_missing_samples_when_remapping: bool,
 ) -> hl.MatrixTable:
     mt = vcf_remap(mt)
-    collected_remap = project_remap_ht.collect()
-    s_dups = [k for k, v in Counter([r.s for r in collected_remap]).items() if v > 1]
+
+    remap_count_by_s = project_remap_ht.aggregate(hl.agg.counter(project_remap_ht.s))
```

Comment:
we can use hail's `aggregate` function to count duplicates without collecting into python lists

Reply:
👍 love the idea!

I think we want to be careful, though, about the performance of calling the `.aggregate()` method twice rather than the single `.collect()`.

I think the idea behind the python Counter here was to take advantage of our knowledge that the number of samples we have is reasonably small (<5k or so) and fits into memory in python without much worry.  in memory python is _way_ faster than hail.

Some local timings:

```
In [19]: %timeit collected_remap = project_remap_ht.collect(); counted = Counter(collected_remap)
5.91 ms ± 320 µs per loop (mean ± std. dev. of 7 runs, 100 loops each)

In [20]: %timeit project_remap_ht.aggregate(hl.agg.counter(project_remap_ht.s))
346 ms ± 8.78 ms per loop (mean ± std. dev. of 7 runs, 1 loop each)

In [21]: %timeit project_remap_ht.aggregate(hl.agg.counter(project_remap_ht.seqr_id))
358 ms ± 30.9 ms per loop (mean ± std. dev. of 7 runs, 1 loop each)
```

Reply:
this all makes sense and wow, it is actually so much faster to use `collect`. I also got those timings with a tiny remap table. I hadn't seen `timeit` before but I'll definitely be using it again.

what was the size of the `project_remap_ht` you used for testing here?